### DISCUSSION TITLE
Greenfield API: In case of 403 Forbidden, tell me what permission I need

### DISCUSSION THREAD

**Discussion by woutersamaey** [btcpayserver/btcpayserver#2636] [Ideas / Feature Requests]
Been developing with Greenfield for a while now and sometimes I run into 403s because of some missing permission.
I'm not sure, but I get the feeling that permissions could be better and that some levels are not 100% what you'd suspect or want to grant.

Can we make it so that Greenfield reports what API permission you need to make the failed request work?

This could go a long way towards more developer-friendliness, especially for newer and less experienced devs.

Is this feasible?

Security wise: I don't see an issue. Knowing which permission you need is a matter that can be found in the docs as well, so no secrets are revealed. Learning your own API rights is also not a secret as we have an endpoint that tells you that already.

**Comment by woutersamaey** [comment]
Can I kindly get your thoughts on this @Kukks and @NicolasDorier ?

**Comment by Kukks** [comment]
I think we can automate this in the API auth handler actually. ACK from my
end. I don't think there are any security or privacy implications with this.

**Reply by woutersamaey** [reply]
Thanks for pointing me in the right direction. Will have a look and try something out.

**Comment by woutersamaey** [comment]
This looks helpful, but need to learn more about events in C#

https://github.com/aspnet/Security/issues/872#issuecomment-389223252
